src/ai/llm/messages/manager.ts

```diff
@@ -295,10 +271,10 @@ export class MessageManager {
      * @returns Formatted system prompt or null/undefined based on formatter implementation
      * @throws Error if formatting fails
      */
-    getFormattedSystemPrompt(): string | null | undefined {
+    async getFormattedSystemPrompt(): Promise<string | null | undefined> {
+        const prompt = await this.getSystemPrompt();
         try {
-            // Check if the formatter implements getSystemPrompt and call it
-            return this.formatter.getSystemPrompt?.(this.systemPrompt);
+            return this.formatter.getSystemPrompt?.(prompt ?? undefined);
```

Comment:
unrelated but we could rename `getSystemPrompt` in `this.formatter` to `formatSystemPrompt` for clarity